packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -197,6 +197,18 @@ def __new__(mcs, name, bases, namespace, **kwargs):
 
         return cls
 
+    @property
+    def server_settings(cls):
+        return cls._server_settings
+
+    @server_settings.setter
+    def server_settings(cls, value):
+        if cls._server_settings is not None:
+            raise AttributeError("'server_settings' attribute was already set")
+        cls._server_settings = value
+
+    _server_settings = None
+
```

Comment:
Can this be defined on `BaseProvider` instead? The metaclass should only be altered if there's a compelling reason to do so.

Reply:
Do you have suggestions on how to define it on `BaseProvider` instead? As far as I know this is not possible.

Reply:
NVM, I see the need for defining this on the metaclass because this is protecting a class attribute and not an instance attribute.